Sources/iron/system/Input.hx

```diff
@@ -175,7 +175,7 @@ class Mouse extends VirtualInput {
 	}
 
 	function buttonIndex(button: String): Int {
-		return button == "left" ? 0 : (button == "right" ? 1 : 2);
+		return button == "left" ? 0 : button == "right" ? 1 : button == "middle" ? 2 : button == "side1" ? 3 : (button == "side2" ? 4 : 2);
```

Comment:
A switch statement is probably more readable here. It's not a big deal though, at least not for me :)

Also, should 2 still be the default value if the `button` name is not in the list of known names? I think we should either return 5 or -1 here (irrespective of introducing a breaking change, which would be pretty minor).

Reply:
Funny story, I actually used switch/case originally to test my buttons as I enjoy them more as well, then back-ported it to the question-mark & colon combination. To answer your questions:
1. Lubos implemented the code so I didn't want to change anything that is preferable, nor did I know if using queston-mark/colon versus switch/case had any particular performance boost. It does however have better white space i.e. 1 line vs 14 lines.
2. `5` would return an extra mouse button, which might raise an error if a user doesn't have that extra mouse button. Better to keep it at `2` (middle mouse) - or `0` (left mouse) if you'd prefer.

Reply:
Oops, good point 👍